core/src/main/java/org/apache/calcite/util/format/PostgresqlDateTimeFormatter.java

```diff
@@ -0,0 +1,660 @@
+            }
+          },
+          "Y"),
+      new NumberFormatPattern(
+          dt -> String.format(Locale.ROOT, "%02d", dt.get(IsoFields.WEEK_OF_WEEK_BASED_YEAR)),
+          "IW"),
+      new NumberFormatPattern(
+          dt -> {
+            final Month month = dt.getMonth();
+            final int dayOfMonth = dt.getDayOfMonth();
+            final int weekNumber = dt.get(IsoFields.WEEK_OF_WEEK_BASED_YEAR);
+
+            if (month == Month.JANUARY && dayOfMonth < 4) {
+              if (weekNumber == 1) {
+                return String.format(Locale.ROOT, "%03d", dt.getDayOfWeek().getValue());
+              }
+            } else if (month == Month.DECEMBER && dayOfMonth >= 29) {
+              if (weekNumber == 1) {
+                return String.format(Locale.ROOT, "%03d", dt.getDayOfWeek().getValue());
+              }
+            }
+
+            return String.format(Locale.ROOT, "%03d",
+                (weekNumber - 1) * 7 + dt.getDayOfWeek().getValue());
+          },
+          "IDDD"),
+      new NumberFormatPattern(
+          dt -> Integer.toString(dt.getDayOfWeek().getValue()),
+          "ID"),
+      new NumberFormatPattern(
+          dt -> {
+            final String yearString = Integer.toString(dt.get(IsoFields.WEEK_BASED_YEAR));
+            return yearString.substring(yearString.length() - 1);
+          },
+          "I"),
+      new StringFormatPattern(
+          dt -> dt.get(ChronoField.ERA) == 0 ? "BC" : "AD",
+          "BC", "AD"),
+      new StringFormatPattern(
+          dt -> dt.get(ChronoField.ERA) == 0 ? "bc" : "ad",
+          "bc", "ad"),
+      new StringFormatPattern(
+          dt -> dt.get(ChronoField.ERA) == 0 ? "B.C." : "A.D.",
+          "B.C.", "A.D."),
+      new StringFormatPattern(
+          dt -> dt.get(ChronoField.ERA) == 0 ? "b.c." : "a.d.",
+          "b.c.", "a.d."),
+      new StringFormatPattern(
+          dt -> {
+            final String monthName =
+                dt.getMonth().getDisplayName(TextStyle.FULL,
+                Locale.getDefault());
+            return monthName.toUpperCase(Locale.ROOT);
+          },
+          "MONTH"),
+      new StringFormatPattern(
+          dt -> {
+            final String monthName =
+                dt.getMonth().getDisplayName(TextStyle.FULL,
+                Locale.getDefault());
+            return monthName.substring(0, 1).toUpperCase(Locale.ROOT)
+                + monthName.substring(1).toLowerCase(Locale.ROOT);
+          },
+          "Month"),
+      new StringFormatPattern(
+          dt -> {
+            final String monthName =
+                dt.getMonth().getDisplayName(TextStyle.FULL,
+                Locale.getDefault());
+            return monthName.toLowerCase(Locale.ROOT);
+          },
+          "month"),
+      new StringFormatPattern(
+          dt -> {
+            final String monthName =
+                dt.getMonth().getDisplayName(TextStyle.SHORT,
+                Locale.ROOT);
```

Comment:
Similar question

Reply:
See my answer above. I'll make the changes.

Reply:
Updated to make use of the TM flag.